gensim/matutils.py

```diff
@@ -332,9 +343,15 @@ def unitvec(vec):
     Output will be in the same format as input (i.e., gensim vector=>gensim vector,
     or numpy array=>numpy array, scipy.sparse=>scipy.sparse).
     """
+    norm = isnorm(norm)
     if scipy.sparse.issparse(vec): # convert scipy.sparse to standard numpy array
         vec = vec.tocsr()
-        veclen = numpy.sqrt(numpy.sum(vec.data ** 2))
+        if norm == 'l0':
+            veclen = len(vec.data)
```

Comment:
Sorry to confuse you, meant scaling in mathematical not IT sense.

What number do we need to divide vector `a=(5,5)` by in order to have `l_0_norm(a)=1`?